notification-portlet-webapp/src/main/webapp/scripts/modal.js

```diff
@@ -0,0 +1,129 @@
+Vue.component("modal", {
+  template: "#modal-template",
+  props: {
+    id: String
+  }
+})
+
+// start app
+new Vue({
+  el: "#notifications-vue-app",
+  data: {
+    items: [{ show: false }],
+    id: "",
+    lastActive: null,
+    bodyOverflow: null
+  },
+  methods: {
+    gotoNext: function(index) {
+      this.items[index].show = false
+
+      const modal = document.getElementById(this.id)
+      let doc = modal && modal.ownerDocument
+      doc.removeEventListener("focus", this.focusListener, true)
+      doc.removeEventListener("keyup", this.keyUpListener)
+
+      if (this.items[index + 1] !== undefined) {
+        this.items[index + 1].show = true
+        this.id = "notification" + index + 2
+        let that = this
+        setTimeout(function() {
+          that.ada(index + 1)
+        }, 250)
+      } else {
+        if (this.lastActive) {
+          this.lastActive.focus()
+        }
+
+        let body = document.getElementsByTagName("body")[0]
+        for (let x = 0; x < body.childElementCount; x++) {
+          body.children[x].removeAttribute("aria-hidden")
+        }
+        body.style.overflow = this.bodyOverflow
+      }
+    },
+    submit: function(notificationId, actionId, index) {
+      const formData = {
+        notificationId: notificationId,
+        actionId: actionId
+      }
+      const formBody = Object.keys(formData)
+        .map(function(key) {
+          return (
+            encodeURIComponent(key) + "=" + encodeURIComponent(formData[key])
+          )
+        })
+        .join("&")
+      this.$http
+        .post(invokeActionUrl, formBody, {
+          credentials: "include",
+          headers: { "Content-Type": "application/x-www-form-urlencoded" }
+        })
+        .then(
+          function(response) {
+            this.gotoNext(index)
+          },
+          function(response) {
+            console.log("Problem posting request")
+          }
+        )
+    },
+    ada: function(index) {
```

Comment:
rather than `ada` could this be named something like `attachListeners`?

Reply:
Sure that is an easy one.